atc/exec/put_step.go

```diff
@@ -220,15 +220,7 @@ func (step *PutStep) run(ctx context.Context, state RunState, delegate PutDelega
 
 	state.StoreResult(step.planID, versionResult)
 
-	step.succeeded = true
-
 	delegate.Finished(logger, 0, versionResult)
 
-	return nil
-
-}
-
-// Succeeded returns true if the resource script exited successfully.
-func (step *PutStep) Succeeded() bool {
-	return step.succeeded
```

Comment:
Took a look at the deepsource output, and sure enough, it pointed out the useful fact that we can remove `step.succeeded` from `PutStep`. Thanks, deepsource!

https://deepsource.io/gh/concourse/concourse/run/970748ed-d236-4715-bd3c-c5ea58a0bc52/go/SCC-U1000